pkg/resources/view_grant.go

```diff
@@ -107,14 +110,23 @@ func CreateViewGrant(data *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	// ID format is <db_name>|<schema_name>|<view_name>|<privilege>
-	// view_name is empty when on_future = true
-	if futureViews {
-		data.SetId(fmt.Sprintf("%v|%v||%v", dbName, schemaName, priv))
-	} else {
-		data.SetId(fmt.Sprintf("%v|%v|%v|%v", dbName, schemaName, viewName, priv))
+	data_identifiers := [][]string{{dbName, schemaName, viewName, ""}}
+
+	if !(futureViews) { // merge with above if/else statement if possible
+		data_identifiers[0][3] = priv
```

Comment:
seems like the thing that changes is if the `viewName` is present or not. but priv is always there

Reply:
@edulop91  

Getting a little frustrated about knowing how to format data_identifiers :/ I'm not sure if I should add priv to the end of dataIdentifiers (new variable name now)?

It's hard for me to make that variable adaptable. This is what I have so far:
```
	dataIdentifiers := make([][]string, 1, 4)
	dataIdentifiers[0][0] = dbName
	dataIdentifiers[0][1] = schemaName
	dataIdentifiers[0][2] = viewName
	dataIdentifiers[0][3] = priv
```
Everything else is the same.